packages/wrangler/src/__tests__/deploy.test.ts

```diff
@@ -4422,6 +4442,243 @@ addEventListener('fetch', event => {});`
 				)
 			);
 		});
+
+		it("should upload an asset manifest of the files in the directory specified by --experimental-assets", async () => {
+			const assets = [
+				{ filePath: "file-1.txt", content: "Content of file-1" },
+				{ filePath: "boop/file-2.txt", content: "Content of file-2" },
+			];
+			writeAssets(assets);
+			writeWorkerSource();
+			const bodies: AssetManifest[] = [];
+			msw.use(
+				http.post<never, AssetManifest>(
+					`*/accounts/some-account-id/workers/scripts/test-name/assets-upload-session`,
+					async ({ request }) => {
+						bodies.push(await request.json());
+						return HttpResponse.json(
+							{
+								success: true,
+								errors: [],
+								messages: [],
+								result: { jwt: "<<aus-completion-token>>", buckets: [] },
+							},
+							{ status: 201 }
+						);
+					}
+				)
+			);
+			// skips asset uploading since empty buckets returned
+			mockSubDomainRequest();
+			mockUploadWorkerRequest({
+				expectedExperimentalAssetsForm: [
+					["metadata", JSON.stringify({ assets: "<<aus-completion-token>>" })],
+				],
+			});
+			await runWrangler(
+				"deploy --name test-name --compatibility-date 2024-07-31 --experimental-assets assets"
+			);
+			expect(bodies.length).toBe(1);
+			expect(bodies[0]).toStrictEqual({
+				manifest: {
+					"/file-1.txt": {
+						hash: "0de3dd5df907418e9730fd2bd747bd5e",
+						size: 17,
+					},
+					"/boop/file-2.txt": {
+						hash: "7574a8cd3094a050388ac9663af1c1d6",
+						size: 17,
+					},
+				},
+			});
+		});
+
+		it("should upload an asset manifest of the files in the directory specified by [experimental_assets] config", async () => {
```

Comment:
And can I also get another which proves out the relative resolution? That the CLI arg is relative to cwd, and that the config is relative to the config file location?